mtg_card_identifier/CardScannerFile.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  CardScannerFile.swift
+//  mtg_card_identifier
+//
+//  Created by mac on 8/16/23.
+//
+
+import SwiftUI
+import UIKit
+import VisionKit
+
+struct CardScanner: UIViewControllerRepresentable {
```

Comment:
Let's make this a bit more generic. Instead of CardScanner, let's call it something like `LiveTextScanner`.
We can make it so the overlay is something we can provide as a parameter:
`LiveTextScanner(scannedText: Binding<[String]>, overlay: UIView? = nil)`
Then I'd make a UIView extension and have a static variable for the `static let cardOverlay: UVIew = ...` so you can call it like:
`LiveTextScanner(scannedText: ..., overlay: .cardOverlay)`
Does that make sense?

Reply:
The `cardOverlay` would pretty much just be the code you have in the makeUIViewController func for the `partialTransparentView`.

Reply:
I guess one downside of this is that it would be harder to do the `regionOfInterest` part when creating the View. 
Another option would be to make an enum for the different kinds of overlay that can be used on it:
```
enum ScannerOverlay { 
case card(cardDimensions: CGSize) 
case .none

var view: UIView? {
switch self {
case .card(let cardDimensions):
let partialTransparentView = ...
return partialTransparentView]
case .none:
return nil
}
}
```
Then this guy would accept one of those:
`LiveTextScanner(scannedText: Binding<[String]>, overlay: ScannerOverlay)`
then you can get the inner rect by doing something like:
```
switch overlay {
case .card(let cardDimensions):
// use cardDimensions
```

One part that might be hard with that is converting from the cardDimensions `CGSize` (width/height) to the `regionOfInterest` CGRect (width/height + x/y of top right corner of the view), but that should  just be maths based on the screen size. Annoying but not too crazy? You'd have to do that maths anyway with the current approach I believe.